### PR TITLE
kafka: upgrade kafka source code and server dependencies to 2.8.1

### PR DESCRIPTION
**Pull request by adamkotwasinski**
Commit Message: kafka: upgrade kafka source code and server dependencies to 2.8.1
Additional Description: as per https://github.com/envoyproxy/envoy/pull/17260#issuecomment-875981315 upgrade the dependencies; also get rid of source patch in favour of processing in python script
There are some minor changes in mesh code due to Kafka protocol spec files changing field _names_ a little (e.g. [this](https://github.com/apache/kafka/blob/2.4.1/clients/src/main/resources/common/message/ProduceRequest.json#L42) vs [this](https://github.com/apache/kafka/blob/2.8.0/clients/src/main/resources/common/message/ProduceRequest.json#L45)), but it does not impact the protocol's functioning (it's the same bytes after all - we just reference them differently).
Risk Level: Low
Testing: integration tests (server versions: 2.4.1, 2.8.1), manual tests for broker-filter (client=2.5.1 + server=2.4.0; client=2.8.1 + server=2.8.1), manual tests for mesh-filter (client=2.4.1; client=2.8.1; server=2.8.1)
Docs Changes: Minor .rst updates (because Kafka versions are explicitly mentioned there)
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by moderation** [comment]
Are you going to bump the dependencies in `bazel/repository_locations.bzl`?

https://github.com/envoyproxy/envoy/blob/main/bazel/repository_locations.bzl#L930-L977

FYI, a release candidate for 2.8.1 came out this morning - https://github.com/apache/kafka/releases/tag/2.8.1-rc0

And bump https://github.com/dpkp/kafka-python/releases/tag/2.0.2

**Comment by adamkotwasinski** [comment]
@moderation 
yes - it's a draft right now, because 2.5 and further ones added new data types so I need to write some more C++ code (uuid, records, float64)
I'd rather do only this one https://github.com/envoyproxy/envoy/blob/main/bazel/repository_locations.bzl#L931 in this PR, because it's going to be noisy anyways ; other ones will follow independently

File: contrib/kafka/filters/network/test/serialization_test.cc [hunk 1]
```diff
@@ -77,9 +79,11 @@ TEST(NullableCompactArrayDeserializer, EmptyBufferShouldNotBeReady) {
 
 TEST_DeserializerShouldDeserialize(Int8Deserializer, int8_t, 42);
 TEST_DeserializerShouldDeserialize(Int16Deserializer, int16_t, 42);
+TEST_DeserializerShouldDeserialize(UInt16Deserializer, uint16_t, 42);
 TEST_DeserializerShouldDeserialize(Int32Deserializer, int32_t, 42);
 TEST_DeserializerShouldDeserialize(UInt32Deserializer, uint32_t, 42);
 TEST_DeserializerShouldDeserialize(Int64Deserializer, int64_t, 42);
+TEST_DeserializerShouldDeserialize(Int64Deserializer, double, 13.25);
```
Comment (adamkotwasinski) [inline contrib/kafka/filters/network/test/serialization_test.cc:86]: typo

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -930,13 +930,13 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Kafka (source)",
         project_desc = "Open-source distributed event streaming platform",
         project_url = "https://kafka.apache.org",
-        version = "2.4.1",
-        sha256 = "740236f44d66e33ea83382383b4fb7eabdab7093a644b525dd5ec90207f933bd",
+        version = "2.8.1-rc1",
+        sha256 = "4f013aadedd0c26c31c67bc8f3984833430316fa990553ded77dfbaa65f43376",
```
Comment (moderation) [inline bazel/repository_locations.bzl:934]: 2.8.1 is out - https://github.com/apache/kafka/releases/tag/2.8.1

**Comment by adamkotwasinski** [comment]
/wait-any
I need to perform some semi-manual tests

**Comment by moderation** [comment]
3.0.0 is out -https://github.com/apache/kafka/releases/tag/3.0.0

**Comment by adamkotwasinski** [comment]
source-code (for generating parser code) and server (for integration tests) dependencies have been updated to 2.8.1
what's coming in the future: update to 3.x (not doing this time to avoid pulling even more noise) and dpkp-kafka-python / librdkafka ones

**Comment by adamkotwasinski** [comment]
/wait - certain name is just wrong

**Comment by adamkotwasinski** [comment]
remove waiting label

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**